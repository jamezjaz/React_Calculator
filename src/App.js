import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h3>REACT CALCULATOR</h3>
      </header>
    </div>
  );
}

export default App;
