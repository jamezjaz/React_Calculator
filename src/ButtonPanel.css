.calculator-grid {
  display: grid;
  justify-content: center;
  min-height: 50vh;
  grid-template-columns: repeat(4, 100px);
}

.calculator-grid > button {
  font-size: 2rem;
}

.span-two {
  grid-column: 1 / 3;
}

.cal-button {
  font-size: 2rem;
}
