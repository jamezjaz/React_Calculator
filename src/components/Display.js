import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Display.css';

const Display = props => {
  const { result } = props;
  return (
    <div className="display">
      <p style={{ fontSize: '2rem', textAlign: 'end' }}>
        {result}
      </p>
    </div>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: 0,
};

export default Display;
