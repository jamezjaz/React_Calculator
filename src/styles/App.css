body {
  overflow-x: hidden;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 700px;
  justify-content: flex-end;
  margin: 0 auto 20px auto;
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(2px + 2vmin);
  color: white;
  margin-bottom: 5px;
}
