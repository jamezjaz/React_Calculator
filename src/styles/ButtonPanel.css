.button-panel {
  display: flex;
  justify-content: center;
  height: 100px;
}
