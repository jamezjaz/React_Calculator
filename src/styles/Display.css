.display {
  display: flex;
  width: 700px;
  background-color: #858693;
  height: 100px;
  color: #fff;
  font-weight: bold;
  padding: 7px;
  justify-content: flex-end;
}
