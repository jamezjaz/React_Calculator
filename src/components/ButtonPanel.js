import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import '../styles/ButtonPanel.css';

const ButtonPanel = props => {
  const buttonGroups = {
    group1: {
      id: 1,
      group: ['AC', '+/-', '%', 'รท'],
    },
    group2: {
      id: 2,
      group: ['7', '8', '9', 'X'],
    },
    group3: {
      id: 3,
      group: ['4', '5', '6', '-'],
    },
    group4: {
      id: 4,
      group: ['1', '2', '3', '+'],
    },
    group5: {
      id: 5,
      group: ['0', '.', '='],
    },
  };

  const { clickHandler } = props;

  return (
    <div>
      <div className="calculator-grid">
        {buttonGroups.group1.group.map(btn => (
          <Button name={btn} key={btn} handleClick={clickHandler} />))}
      </div>
      <div className="calculator-grid">
        {buttonGroups.group2.group.map(btn => (
          <Button name={btn} key={btn} handleClick={clickHandler} />))}
      </div>
      <div className="calculator-grid">
        {buttonGroups.group3.group.map(btn => (
          <Button name={btn} key={btn} handleClick={clickHandler} />))}
      </div>
      <div className="calculator-grid">
        {buttonGroups.group4.group.map(btn => (
          <Button name={btn} key={btn} handleClick={clickHandler} />))}
      </div>
      <div className="calculator-grid">
        {buttonGroups.group5.group.map(btn => (
          <Button name={btn} key={btn} handleClick={clickHandler} className={btn === '0' ? 'span-two' : null} />))}
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
