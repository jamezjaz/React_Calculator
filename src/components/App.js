import '../App.css';
import Button from './Button';
import ButtonPanel from './ButtonPanel';
import Display from './Display';

function App() {
  return (
    <>
      <div className="App">
        <header className="App-header">
          <h3>REACT CALCULATOR</h3>
        </header>
        <Display result="5 * 5" />
        <ButtonPanel />
        <Button name="Calculator" />
      </div>
    </>
  );
}

export default App;
