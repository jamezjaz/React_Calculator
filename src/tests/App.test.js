import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });
import App from '../components/App';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';

it('it renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<App />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders Display correctly', () => {
    expect(wrapper.find(Display)).toHaveLength(1);
  });

  it('renders ButtonPanel correctly', () => {
    expect(wrapper.find(ButtonPanel)).toHaveLength(1);
  });
});