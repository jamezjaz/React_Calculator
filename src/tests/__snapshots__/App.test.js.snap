// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
Array [
  <header
    className="App-header"
  >
    <h3>
      REACT CALCULATOR
    </h3>
  </header>,
  <div
    className="App"
  >
    <div
      className="display"
    >
      <p
        style={
          Object {
            "fontSize": "2rem",
            "textAlign": "end",
          }
        }
      >
        0
      </p>
    </div>
    <div>
      <div
        className="button-panel"
      >
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className="orange"
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </p>
        </div>
      </div>
      <div
        className="button-panel"
      >
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className="orange"
              onClick={[Function]}
              type="button"
            >
              X
            </button>
          </p>
        </div>
      </div>
      <div
        className="button-panel"
      >
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className="orange"
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </p>
        </div>
      </div>
      <div
        className="button-panel"
      >
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className="orange"
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </p>
        </div>
      </div>
      <div
        className="button-panel zero"
      >
        <div>
          <p>
            <button
              className=" wide"
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className=""
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </p>
        </div>
        <div>
          <p>
            <button
              className="orange"
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
